packages/selfie-segmentation/ios/Plugin/Classes/ProcessImageResult.swift

```diff
@@ -1,47 +1,32 @@
 import Foundation
 import Capacitor
-import MLKitVision
-import MLKitSegmentationSelfie
 
 @objc class ProcessImageResult: NSObject {
-    let segmentationMask: SegmentationMask
+    let image: UIImage
 
-    init(segmentationMask: SegmentationMask) {
-        self.segmentationMask = segmentationMask
+    init(image: UIImage) {
+        self.image = image
     }
 
     func toJSObject() -> JSObject {
-        let (maskResult, maskWidth, maskHeight) = createMaskResult(mask: segmentationMask)
-
         var result = JSObject()
-        result["mask"] = maskResult
-        result["width"] = maskWidth
-        result["height"] = maskHeight
-
-        return result
-    }
 
-    private func createMaskResult(mask: SegmentationMask) -> (JSArray, Int, Int) {
-        var result = JSArray()
+        if let data = image.pngData() {
+            do {
+                let path = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask)[0]
+                let name = "photo-"+UUID().uuidString+".png"
+                let url = path.appendingPathComponent(name)
+                try data.write(to: url)
 
-        let maskWidth = CVPixelBufferGetWidth(mask.buffer)
-        let maskHeight = CVPixelBufferGetHeight(mask.buffer)
-
-        CVPixelBufferLockBaseAddress(mask.buffer, CVPixelBufferLockFlags.readOnly)
-        let maskBytesPerRow = CVPixelBufferGetBytesPerRow(mask.buffer)
-        var maskAddress =
-            CVPixelBufferGetBaseAddress(mask.buffer)!.bindMemory(
-                to: Float32.self, capacity: maskBytesPerRow * maskHeight)
-
-        for _ in 0...(maskHeight - 1) {
-            for col in 0...(maskWidth - 1) {
-                // Gets the confidence of the pixel in the mask being in the foreground.
-                let foregroundConfidence: Float32 = maskAddress[col]
-                result.append(foregroundConfidence)
+                result["path"] = url.absoluteString
```

Comment:
@robingenz that's the part I am talking about in the description. The resulting path looks like a valid path to the image file, but when I use it in the web app to show the image, it doesn't work. Do you have an idea what I am doing wrong here?

Reply:
You plugin code was fine. I just refactored it a bit. It seems like this missing line was the issue: https://github.com/trancee/capacitor-mlkit-plugin-demo/blob/7c0af5afffac9e61f63964f87d425f20e85eb0c2/src/app/modules/selfie-segmentation/selfie-segmentation.page.ts#L55
I've fixed it.